File: src/sorting/SelectionSort.java
package sorting;

public class SelectionSort {

	void selectionSort(int[] arr) {
		int size = arr.length;
		int min, temp, i, j;
		for (i = 0; i < size - 1; i++) {
			min = i;
			for (j = i + 1; j < size; j++) {
				if (arr[min] > arr[j]) {
					min = j;
				}
			}
			// swap elements
			temp = arr[min];
			arr[min] = arr[i];
			arr[i] = temp;
		}
	}
}

File: src/main/kotlin/algos/arrays/sorting/SelectionSort.kt
package algos.arrays.sorting

class SelectionSort {

    fun sort(array: Array<Int>) {
        array.forEachIndexed { index, _ ->
            val smallest = smallestIndex(array, index)
            swap(array, index, smallest)
        }
    }

    private fun smallestIndex(array: Array<Int>, start: Int): Int {
        var smallest = start

        (start..array.lastIndex).forEach { i ->
            if (array[i] < array[smallest])
                smallest = i
        }

        return smallest
    }

    private fun swap(array: Array<Int>, i: Int, j: Int) {
        val temp = array[i]
        array[i] = array[j]
        array[j] = temp
    }
}

fun main() {
    val arr = arrayOf(4, 1, 6, 5, 2, 3)
    println(arr.contentToString())

    SelectionSort().sort(arr)

    println("\nsorted:")
    println(arr.contentToString())
}